syntax = "proto3";

enum MxStoreType {
  LOG = 0;
  CMD = 1;
  GPIO = 2;
  ADS1015 = 3;
  MCP2515 = 4;
  ADS1259 = 5;
  ADT7476A = 6;
  ADC = 7;
  PCA9539R = 8;
  MCP23008 = 9;
  END = 10;
}

enum MxCmdType {
  NOP = 0;
  FINISH_INIT_CONDS = 1;
  NUM_CMDS = 3;
}

message MxLog {
  bytes log = 1;
}

message MxCmd {
  MxCmdType cmd = 1;
}

message MxStoreInfo {
  uint64 key = 1;
  MxStoreType type = 2;
  bytes msg = 3;
}

message MxStoreUpdate {
  uint64 key = 1;
  MxStoreType type = 2;
  bytes msg = 3;
  bytes mask = 4;
}
